sui_core/src/consensus_adapter.rs

```diff
@@ -129,91 +129,91 @@ impl ConsensusAdapter {
         }
 
         // Wait for the consensus to sequence the certificate and assign locks to shared objects.
-        timeout(self.max_delay, receiver)
-            .await
-            .map_err(|e| SuiError::ConsensusConnectionBroken(e.to_string()))?
-            .expect("Channel with consensus listener dropped")
+        match timeout(self.max_delay, receiver).await {
+            Ok(reply) => reply.expect("Channel with consensus listener dropped"),
+            Err(e) => {
+                let message = ConsensusListenerMessage::Cleanup(serialized);
+                self.tx_consensus_listener
+                    .send(message)
+                    .await
+                    .expect("Channel with consensus listener dropped");
```

Comment:
1. Nit: we're reusing the same message in two places. Local var?
2. Could we also use `warn!` logging under a suitable span?

Reply:
It is not the same, once it is `New` the other is `Cleanup`

Reply:
Sorry, I meant error message: "Channel with consensus listener dropped"

Reply:
ah ok, made clearer error messages (they could be made more specific)